### PR TITLE
feat(protocol): Move proofTimeTarget to state var and adjust scripts/tests

### PR DESCRIPTION
**Pull request by adaki2004**
proofTImeTarget is now state var. Tests / scripts are adjusted.

Will add setter for them (proofTImeTarget + proofTImeIssued) soon - after lunch.

### CODE REVIEW

File: packages/protocol/contracts/L1/TaikoData.sol [hunk 1]
```diff
@@ -139,7 +139,7 @@ library TaikoData {
         // Slot 7: never or rarely changed
         uint64 genesisHeight;
         uint64 genesisTimestamp;
-        uint64 __reserved71;
+        uint64 proofTimeTarget;
```
Comment (dantaik) [inline packages/protocol/contracts/L1/TaikoData.sol:142]: It seems using `__reserved91` for `proofTimeTarget` will save some gas as neither `genesisHeight` or `genesisTimestamp` is accessed when block is proposed/proven.

File: packages/protocol/script/DeployOnL1.s.sol [hunk 2]
```diff
@@ -44,10 +44,18 @@ contract DeployOnL1 is Script {
 
     uint256 public taikoTokenPremintAmount = vm.envUint("TAIKO_TOKEN_PREMINT_AMOUNT");
 
+    // Change it based on 'consensus' / experience / expected result
+    // Based in seconds. Please set carefully.
+    // For testnet it could be somewhere 85-100s
+    // For mainnet it could be around 1800 s (30mins)
+    // Can be adjusted later with setters
+    uint64 public INITIAL_PROOF_TIME_TARGET = 0;
```
Comment (dantaik) [inline packages/protocol/script/DeployOnL1.s.sol:52]: please also read from env file as other input variables.

File: packages/protocol/script/DetermineNewProofTimeIssued.s.sol [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+pragma solidity ^0.8.18;
+
+import "forge-std/Script.sol";
+import "forge-std/console2.sol";
+import {LibLn} from "../test/LibLn.sol";
+
+uint16 constant DESIRED_PROOF_TIME_TARGET = 500;
+uint8 constant ADJUSTMENT_QUOTIENT = 16;
+
+contract DetermineProofTImeIssued is Script {
```
Comment (dantaik) [inline packages/protocol/script/DetermineNewProofTimeIssued.s.sol:11]:
```suggestion
contract DetermineProofTimeIssued is Script {
```

File: packages/protocol/contracts/L1/TaikoL1.sol [hunk 4]
```diff
@@ -38,13 +38,15 @@ contract TaikoL1 is EssentialContract, ICrossChainSync, TaikoEvents, TaikoErrors
...
      *        with the initial block fee.
      */
     function init(
         address _addressManager,
         bytes32 _genesisBlockHash,
         uint64 _initBlockFee,
+        uint64 _initProofTimeTarget,
         uint64 _initProofTimeIssued
     ) external initializer {
         EssentialContract._init(_addressManager);
```
Comment (dantaik) [inline packages/protocol/contracts/L1/TaikoL1.sol:52]: Do you not want to add a function `setProofTimeTarget() onlyOwner`?

**Review by dantaik — CHANGES_REQUESTED** [review]

**Comment by dantaik** [comment]
> Will add setter for them (proofTImeTarget + proofTImeIssued) soon - after lunch.

Maybe we should add them now :)

File: packages/protocol/contracts/L1/TaikoL1.sol [hunk 5]
```diff
@@ -32,19 +32,23 @@ contract TaikoL1 is EssentialContract, ICrossChainSync, TaikoEvents, TaikoErrors
         depositEtherToL2();
     }
 
+    event ProofTimeTargetChanged(uint64 proofTimeTarget);
```
Comment (davidtaikocha) [inline packages/protocol/contracts/L1/TaikoL1.sol:35]: let also add this event to `TaikoEvent.sol`?

**Review by Brechtpd — APPROVED** [review]

**Review by davidtaikocha — APPROVED** [review]

**Review by adaki2004 — COMMENTED** [review]
Resolved requests.

**Review by dantaik — APPROVED** [review]

**Pull request merged**